EventListener/ReportSubscriber.php

```diff
@@ -0,0 +1,268 @@
+<?php
+
+namespace MauticPlugin\LeuchtfeuerCompanySegmentsBundle\EventListener;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\Expression\CompositeExpression;
+use Mautic\CoreBundle\Helper\InputHelper;
+use Mautic\LeadBundle\Model\CompanyReportData;
+use Mautic\ReportBundle\Event\ReportBuilderEvent;
+use Mautic\ReportBundle\Event\ReportGeneratorEvent;
+use Mautic\ReportBundle\ReportEvents;
+use MauticPlugin\LeuchtfeuerCompanySegmentsBundle\Entity\CompanySegmentRepository;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class ReportSubscriber implements EventSubscriberInterface
+{
+    public const CONTEXT_COMPANY_SEGMENTS         = 'company_segments';
+    public const COMPANY_TABLE                    = 'companies';
+    public const COMPANIES_PREFIX                 = 'comp';
+    public const COMPANY_SEGMENTS_XREF_PREFIX     = 'csx';
+    public const COMPANY_SEGMENTS_XREF_TABLE      = 'company_segment_xref';
```

Comment:
Please check your implementation works in latest version of the bundle. (there are no `company_segment_xref` table anymore.

Reply:
This one seems to be outdated. CompaniesSegments::TABLE_NAME is already used in the current code

Reply:
For the companies table I tried to use a constant from the entity. But it's hard coded in the entity, so I saw no other possibility than sticking with the hard coded way